app/component/itinerary/navigator/NaviCardContainer.js

```diff
@@ -94,30 +96,29 @@ function NaviCardContainer(
       ),
     );
 
-    if (currentLeg) {
-      if (nextLeg?.transitLeg) {
-        // Messages for NaviStack.
-        addMessages(incomingMessages, [
-          ...getTransitLegState(nextLeg, intl, messages, time),
-          ...getAdditionalMessages(nextLeg, time, intl, config, messages),
-        ]);
-      }
-      if (legChanged) {
-        focusToLeg?.(currentLeg);
-        setCardExpanded(false);
-      }
+    if (nextLeg?.transitLeg) {
+      // Messages for NaviStack.
+      addMessages(incomingMessages, [
+        ...getTransitLegState(nextLeg, intl, messages, time),
+        ...getAdditionalMessages(nextLeg, time, intl, config, messages),
+      ]);
+    }
+
+    if (currentLeg && legChanged) {
+      focusToLeg?.(currentLeg);
+      setCardExpanded(false);
     }
     if (incomingMessages.size || legChanged) {
       // Handle messages when new messages arrives.
 
       // Current active messages. Filter away expired messages.
       const previousValidMessages = legChanged
-        ? activeMessages.filter(m => m.expiresOn < time)
+        ? activeMessages.filter(m => !m.expiresOn || m.expiresOn > time)
         : activeMessages;
 
       // handle messages that are updated.
       const keptMessages = previousValidMessages.filter(
-        msg => !!incomingMessages.get(msg.id),
+        msg => !incomingMessages.get(msg.id),
```

Comment:
Is this 180° turn in the logic completely intentional?
![image](https://github.com/user-attachments/assets/aa5d097d-1419-46f3-b37c-9fb711edaa09)

Reply:
Yes, I made a bug earlier. It is hard to remember if true keeps or removes :)